From 2fa10bd59f5dd2a8481fbe63c14daff1ebe9f53d Mon Sep 17 00:00:00 2001
From: SeongJae Park <sj@kernel.org>
Date: Mon, 29 Apr 2024 17:16:56 -0700
Subject: [PATCH] Docs/mm/damon/maintainer-profile: introduce HacKerMaiL

DAMON community uses the mailing list based workflow.  Some people
argues that's not very easy for beginners.  Even some experienced
developers get trouble from the workflow such as handling of the large
volume of the mails.  HacKerMaiL (hkml) [1] is a tool that DAMON
maintainer is developing and maintaining for his efficient LKML
management.  The tool is still under active development, but matured
enough for small projects like DAMON, and being successfully used by not
only the maintainer but also some other developers.

And because the maintainer of the tool and DAMON is the same person, the
tool could be a good option for DAMON contributors.  Introduce the tool
as one of such options on DAMON maintainer's profile document.

[1] https://github.com/sjp38/hackermail

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 Documentation/mm/damon/maintainer-profile.rst | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/Documentation/mm/damon/maintainer-profile.rst b/Documentation/mm/damon/maintainer-profile.rst
index 8213cf61d38a..24a373c39c44 100644
--- a/Documentation/mm/damon/maintainer-profile.rst
+++ b/Documentation/mm/damon/maintainer-profile.rst
@@ -53,6 +53,19 @@ Mon-Fri) in PT (Pacific Time).  The response to patches will occasionally be
 slow.  Do not hesitate to send a ping if you have not heard back within a week
 of sending a patch.
 
+Mailing tool support
+--------------------
+
+DAMON uses the mailing list based workflow.  For easy DAMON development
+participation, DAMON maintainer is also developing and maintaining a simple
+tool for the workflow, called HacKerMaiL (hkml) [8]_ .  Because of the purpose
+and the fact that the maintainer of the tool and DAMON is the same person, the
+tool will support DAMON development workflow use cases as much as possible.
+
+If you want to participate in DAMON development but having troubles due to the
+workflow or existing tools, you can try hkml, report any issue from it, and
+request new features to the maintainer.
+
 
 .. [1] https://git.kernel.org/akpm/mm/h/mm-unstable
 .. [2] https://git.kernel.org/sj/h/damon/next
@@ -61,3 +74,4 @@ of sending a patch.
 .. [5] https://github.com/awslabs/damon-tests/blob/master/corr/tests/kunit.sh
 .. [6] https://github.com/awslabs/damon-tests/tree/master/corr
 .. [7] https://github.com/awslabs/damon-tests/tree/master/perf
+.. [8] https://github.com/sjp38/hackermail
-- 
2.39.2

